torch_geometric/distributed/partition.py

```diff
@@ -85,6 +85,32 @@ def __init__(
     def is_hetero(self) -> bool:
         return isinstance(self.data, HeteroData)
 
+    @property
+    def is_node_level_time(self) -> bool:
+        if 'time' in self.data:
+            if self.is_hetero:
+                for store in self.data.node_stores:
+                    if 'time' in store:
+                        return True
+            else:
+                if self.data.is_node_attr('time'):
+                    return True
+        return False
+
+    @property
+    def is_edge_level_time(self) -> bool:
+        if 'edge_time' in self.data:
+            return True
+        if 'time' in self.data:
+            if self.is_hetero:
+                for store in self.data.edge_stores:
+                    if 'time' in store:
+                        return True
+            else:
+                if self.data.is_edge_attr('time'):
+                    return True
```

Comment:
Can't we use `is_edge_attr` both for hetero/homo case?

Reply:
I am receiving _KeyError: 'time'_ when I check if the 'time' is an edge attribute.

Reply:
It was solved in the same way in the `NeighborSampler.init()`: [link](https://github.com/pyg-team/pytorch_geometric/blob/32e3872721b9475c760d54a2c22ca8365f957017/torch_geometric/sampler/neighbor_sampler.py#L110C15-L110C15)